lab_5_scrapper/scrapper.py

```diff
@@ -140,10 +290,74 @@ def get_search_urls(self) -> list:
         Returns:
             list: seed_urls param
         """
+        return self.config.get_seed_urls()
 
 
 # 10
 # 4, 6, 8, 10
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler is a child of Crawler class.
+    """
+    def __init__(self, config: Config) -> None:
+        super().__init__(config)
+
+        self.start_url = self.base_url
+
+        self.already_crawled_path = constants.ASSETS_PATH.parent / 'already_crawled_urls.txt'
+        with open(self.already_crawled_path, 'r', encoding='utf-8') as f:
+            self.already_crawled = set(f.readlines())
+
+        self.article_tag = 'ARTICLE URL: '
```

Comment:
strange usage scenario

Reply:
I used it to differentiate article links while saving every url from page to file, so I would not search for them again if we return after a crash.

Reply:
this strange decision is because of strange format design for storing urls history. if it were a more flexible format (for example, JSON) you won't need to add artificial prefixes for lines. rework in any convenient manner, but avoid prefixes

Reply:
Reworked as separate files